src/connection.js

```diff
@@ -860,7 +916,18 @@ class Connection extends EventEmitter {
     this.debug.payload(function() {
       return preloginPayload.toString('  ');
     });
-
+    if (this.fedAuthInfo.method != undefined) {
+      if (0 != preloginPayload.fedAuthRequired && 1 != preloginPayload.fedAuthRequired) {
+        this.emit('connect', ConnectionError('Server sent an unexpected value for FedAuthRequired PreLogin Option. Value was' + preloginPayload.fedAuthRequired, 'EFEDAUTH'));
```

Comment:
Can you use template strings for error message? It will be easier to keep track of spaces while appending variables. 
Also, I think this message is too technical for users to understand 😅 `FedAuthRequired PreLogin Option`